lib/plugins/aws/package/compile/events/websockets/lib/validate.js

```diff
@@ -78,9 +78,13 @@ module.exports = {
             } else if (_.isObject(event.websocket.authorizer)) {
               websocketObj.authorizer = {};
               if (event.websocket.authorizer.arn) {
-                websocketObj.authorizer.name = getAuthorizerNameFromArn(
-                  event.websocket.authorizer.arn
-                );
+                if (event.websocket.authorizer.name) {
+                  websocketObj.authorizer.name = event.websocket.authorizer.name;
+                } else {
+                  websocketObj.authorizer.name = getAuthorizerNameFromArn(
```

Comment:
As now we will support non string form of ARN, it'll be good to throw here meaningful error if `arn` is not a string (like: `Non-string "arn" setting need to be accompanied with "name"`)